sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -491,10 +491,30 @@ case class GpuCast(
   }
 
   private def castTimestampToString(input: GpuColumnVector) = {
-    GpuColumnVector.from(
-      withResource(input.getBase.asStrings("%Y-%m-%d %H:%M:%S.%3f")) { cv =>
-        cv.stringReplaceWithBackrefs(GpuCast.TIMESTAMP_TRUNCATE_REGEX,"\\1\\2\\3")
-      })
+    // https://github.com/rapidsai/cudf/issues/5166
+    // The time is off by 1 second if the result is < 0
+    val adjustedTimestamp = withResource(input.getBase
+        .castTo(DType.TIMESTAMP_MICROSECONDS)) { micros =>
+      withResource(micros.castTo(DType.INT64)) { micros =>
+        withResource(Scalar.fromLong(1000000)) { oneSecond =>
+          withResource(micros.sub(oneSecond)) { plusOne =>
```

Comment:
nit: I don't think this is `plusOne` when we subtract one second from it.

Reply:
oops .. thanks, fixed.